droidlet/perception/robot/perception.py

```diff
@@ -98,14 +96,16 @@ def perceive(self, rgb_depth, xyz, previous_objects, force=False):
             old_image, detections, humans, old_xyz = None, None, None, None
 
         self.log(rgb_depth, detections, humans, old_image)
+        perceive_output = namedtuple("perception", ["new_detections", "updated_detections", "humans"])
```

Comment:
you might want to lift this above, i.e. maybe right above the class `Perception` (in line 20), or maybe make it a static member of Perception -- or another thought is to move it to shared_data_structs.py. Also, it's convenient to add default values to the namedtuple.

For example:

```python
PerceptionData = namedtuple("perception_output", ["humans", "new_objects", "updated_objects"], defaults=[None, None, None])
```

Then, you'd do something like:

```python
return PerceptionOutput(humans, new_detections, updated_detections)
```

That will simplify the logic in `loco_memory.py`, where you currently have something like this:

```python
 if hasattr(perception_output, "new_detections") and perception_output.new_detections:
```

it becomes

```python
 if perception_output.new_detections is not None:
```

Similarly / additionally, in the tests, you can create empty or partial `PerceptionOutput()`s instead of `namedtuple()` as needed for testing.

Reply:
Ah yes yes, I am kicking myself for not putting this in `shared_data_structs` with defaults.
Thank you for the very patient review - pushing this fix out